src/d3a/models/strategy/storage.py

```diff
@@ -15,208 +15,202 @@
+                                 final_rate=final_buying_rate,
+                                 fit_to_limit=fit_to_limit,
+                                 energy_rate_change_per_update=-1 * energy_rate_change_per_update,
+                                 update_interval=update_interval)
+        self.state = \
+            StorageState(initial_soc=initial_soc,
+                         initial_energy_origin=initial_energy_origin,
+                         capacity=battery_capacity_kWh,
+                         max_abs_battery_power_kW=max_abs_battery_power_kW,
+                         loss_per_hour=0.0,
+                         min_allowed_soc=min_allowed_soc)
         self.cap_price_strategy = cap_price_strategy
         self.balancing_energy_ratio = BalancingRatio(*balancing_energy_ratio)
 
-    def area_reconfigure_event(self, risk=None, initial_rate_option=None,
-                               energy_rate_decrease_option=None,
-                               energy_rate_decrease_per_update=None,
-                               battery_capacity_kWh=None,
-                               max_abs_battery_power_kW=None, break_even=None,
-                               min_allowed_soc=None):
-        if break_even is not None:
-            self.break_even = self._validate_break_even_points(break_even)
-            self.initial_selling_rate = list(break_even.values())[0][1]
-            self.max_buying_rate_profile = {k: v.sell for k, v in break_even.items()}
-
-        self._validate_constructor_arguments(risk, None, None, battery_capacity_kWh,
-                                             min_allowed_soc, self.initial_selling_rate)
-        self.assign_offermixin_arguments(initial_rate_option, energy_rate_decrease_option,
-                                         energy_rate_decrease_per_update)
-        if battery_capacity_kWh is not None:
-            self.state.capacity = battery_capacity_kWh
-        if max_abs_battery_power_kW is not None:
-            self.state.max_abs_battery_power_kW = max_abs_battery_power_kW
-        if risk is not None:
-            self.risk = risk
-        if min_allowed_soc is not None:
-            self.state.min_allowed_soc = min_allowed_soc
+    def area_reconfigure_event(self, cap_price_strategy=None,
+                               initial_selling_rate=None, final_selling_rate=None,
+                               initial_buying_rate=None, final_buying_rate=None,
+                               fit_to_limit=None, update_interval=None,
+                               energy_rate_change_per_update=None):
+
+        self._validate_constructor_arguments(
+            initial_selling_rate=initial_selling_rate,
+            final_selling_rate=final_selling_rate,
+            initial_buying_rate=initial_buying_rate,
+            final_buying_rate=final_buying_rate,
+            energy_rate_change_per_update=energy_rate_change_per_update)
+        if cap_price_strategy is not None:
```

Comment:
Because this is getting lengthy, I would suggest to gather all these read_arbitrary_profile calls in a separate method, named e.g. populate_offer_bid_update